blueprints/mods/token.lua

```diff
@@ -0,0 +1,240 @@
+          if the Cast tag is not set on the TransferFrom message
+        ]]
+      --
+      if not msg.Cast then
+        -- Send Debit-Notice to the Owner
+        ao.send({
+          Target = msg.Owner,
+          Action = 'Debit-Notice',
+          Recipient = msg.Recipient,
+          Quantity = tostring(qty),
+          Data = Colors.gray ..
+              "You transferred " ..
+              Colors.blue .. msg.Quantity .. Colors.gray .. " to " .. Colors.green .. msg.Recipient .. Colors.reset
+        })
+        -- Send Credit-Notice to the Recipient
+        ao.send({
+          Target = msg.Recipient,
+          Action = 'Credit-Notice',
+          Sender = msg.Owner,
+          Quantity = tostring(qty),
+          Data = Colors.gray ..
+              "You received " ..
+              Colors.blue .. msg.Quantity .. Colors.gray .. " from " .. Colors.green .. msg.From .. Colors.reset
+        })
+      end
+    else
+      ao.send({
+        Target = msg.Owner,
+        Action = 'TransferFrom-Error',
+        ['Message-Id'] = msg.Id,
+        Error = 'Insufficient Balance!'
+      })
+    end
+  else
+    ao.send({
+      Target = msg.Owner,
+      Action = 'TransferFrom-Error',
+      ['Message-Id'] = msg.Id,
+      Error = 'Insufficient Allowance!'
+    })
+  end
+end
+
+function Mod.approve(msg)
+  assert(type(msg.Spender) == 'string', 'Spender is required!')
+  assert(type(msg.Quantity) == 'string', 'Quantity is required!')
+  assert(bint.__lt(0, bint(msg.Quantity)), 'Quantity must be greater than 0')
+  if not Allowances[msg.From] then Allowances[msg.From] = {} end
+  Allowances[msg.From][msg.Spender] = msg.Quantity
+end
+
+function Mod.mint(msg)
+  assert(msg.from == Minter, 'Not Authorized')
+  assert(type(msg.Quantity) == 'string', 'Quantity is required!')
+  assert(bint.__lt(0, msg.Quantity), 'Quantity must be greater than zero!')
+
+  if not Balances[ao.id] then Balances[ao.id] = "0" end
+
+  if msg.From == ao.id then
+    -- Add tokens to the token pool, according to Quantity
+    Balances[msg.From] = tostring(bint.__add(Balances[Owner], msg.Quantity))
+    ao.send({
+      Target = msg.From,
+      Data = Colors.gray .. "Successfully minted " .. Colors.blue .. msg.Quantity .. Colors.reset
+    })
+  else
+    ao.send({
+      Target = msg.From,
+      Action = 'Mint-Error',
+      ['Message-Id'] = msg.Id,
+      Error = 'Only the Process Owner can mint new ' .. Ticker .. ' tokens!'
+    })
+  end
+end
+
+function Mod.setMinter(msg)
+  --[[
+          Allows the Minter to set another processId as the Minter
+          If the Minter wants to prevent any future mints they can set the processId to the processId of the token
+        ]]
+      --
+  assert(msg.from == Minter, 'Not Authorized')
+  assert(type(msg.Minter) == 'string', 'Minter is required!')
+  Minter = msg.Minter
+end
+
+function Mod.init(msg)
+  assert(type(msg.Minter) == 'string', 'Minter is required!')
```

Comment:
Could be good to put a call to `ao.isTrusted(msg)` here, so that only the process owner can invoke this handler.

https://github.com/permaweb/ao/blob/main/dev-cli/container/src/ao.lua#L171

Reply:
I added `ao.isTrusted(msg)` to the init method and renamed the `Owner` prop on the msg object to `OwnerBalance` since it seems `Owner` is a reserved name

Reply:
Yeah, that's true. `Owner` is a reserved word on the top level message. However, I think that `msg.Tags.Owner` would work. Still, your edit might be better.